network/netcore/src/transport/tcp.rs

```diff
@@ -38,6 +38,13 @@ pub struct TcpTransport {
 }
 
 impl TcpTransport {
+    pub fn new() -> Self {
+        Self {
+            nodelay: Some(true),
+            ..Self::default()
+        }
+    }
+
```

Comment:
After thinking a bit more about this: `netcore` is meant to be libra-independent. Even better would be to add something like

```rust
// network/src/transport.rs
const LIBRA_TCP_TRANSPORT: tcp::TcpTransport = tcp::TcpTransport {
    // Use default options.
    recv_buffer_size: None,
    send_buffer_size: None,
    ttl: None,
    keepalive: None,
    // Use TCP_NODELAY for libra tcp connections.
    nodelay: Some(true),
};
```

and then the functions would use

```rust
LIBRA_TCP_TRANSPORT
    .and_then(/* .. */)
```

Reply:
Updated